devex/vsextension/vsix-release-process.md

```diff
@@ -0,0 +1,78 @@
+# VS Extension Release Process
+
+The process used to build and release the nuget package from these
+sources is as follows.  Currently the signature on the release is
+a Microsoft signature generated using the Microsoft [PODSS](https://aka.ms/podss) process.
+In the future, this process will change to a use a non-Microsoft
+signature once another signing process is defined for Open Enclave.
+
+1. Build the latest [Cross-Platform Open Enclave SDK NuGet Package](https://github.com/HernanGatta/openenclave-cross-plat-nuget).
+   Make a note of the new version number for use in step 3.
+
+2. Make a note of the old version referenced in the VS extension sources.
+   This can be found by running the following in the vsextension directory:
+
+      ```cmd
+      findstr open-enclave-cross ProjectWizard\source.extension.vsixmanifest
+      ```
+
+3. Update the open-enclave-cross version referenced in the VS extension
+   sources. For example, from a bash prompt, to update from
+   0.8.1-c3b6262c-3 to 0.11.0-rc1-cbe4dedc-1:
+
+      ```bash
+      grep -RiIl '0.8.1-c3b6262c-3' | xargs sed -i 's/0.8.1-c3b6262c-3/0.11.0-rc1-cbe4dedc-1/g'
+      ```
+
+4. Bump the Identity Version number in the ProjectWizard\source.extension.vsixmanifest
+   file (0.7.29 in the example below). By convension, the major and minor version
+   number should match those of open-enclave-cross, and the patch
+   number should be monotonically increasing (e.g., go from 0.7.29 to 0.8.30).
+
+      ```
+      <Identity Id="OpenEnclaveVisualStudioExtension-1" Version="0.7.29" Language="en-US" Publisher="Microsoft" />
+      ```
+
+5. Open VSExtension.sln in VS2019 and run code analysis on the solution
+   (Analyze -> Run Code Analysis -> On Solution) and verify there's no
+   errors or warnings.
+
+6. Build the Release configuration.
+
+7. Copy the vsix file to a new directory (e.g., a "publish" directory under
+   the Release directory).  Make sure there's nothing else in that directory.
+
+8. Generate a PODSS codesign request that references that directory as follows:
```

Comment:
Is PODSS an internal MSFT tool?

Reply:
Yes.  It's the tool used to generate a Microsoft signature.  Eventually we want a non-Microsoft signature just like the rest of the OE SDK should have, but right now it's a Microsoft signature like the OE SDK is.  We should move them both at the same time.